openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,490 @@
+* You can check your code with the tool  `tools/checker.php`. Call `php tools/checker.php` for detailed instructions.
+
+Please open a pull request and follow the [checklist](pull-request-checklist.md).
+
+### How do I update the [openms.de](https://www.openms.de)?
+
+Login to the [wordpress admin area](www.openms.de/wp-admin) with your username and password assigned by the current Homepage maintainers.
+
+### Can I use QT designer to create GUI widgets?
+
+Yes. If you want to create a class called `Widget: Create .ui-File` with `QT designer` and store it as `Widget.ui.`, add the class to  `sources.cmake`.
+From the .ui-File the file `include/OpenMS/VISUAL/UIC/ClassTemplate.h` is generated by the build system.
+DO NOT CHECK-IN THIS FILE, AS IT IS GENERATED AUTOMATICALLY, WHEN NEEDED!!!
+Derive the class `Widget` from `WidgetTemplate`. You need to check in the `Widget.h` and `Widget.cpp` files.
+
+### Can the START_SECTION-macro not handle template methods that have two or more arguments?
+
+Put round brackets around the method declaration. Then it should work.
+
+### Where can I find the binary installers created?
+
+View the binary installers at the [build archive](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/OpenMSInstaller/nightly/).
+Please verify the creation date of the individual installers, as there may have been an error while creating the installer.
+
+## Troubleshooting
+
+The following section provides information about how to troubleshoot common OpenMS issues.
+
+### OpenMS complains about boost not being found but I'm sure its there!
+
+`CMake` got confused. Set up a new build directory and try again. If you build in-source (not recommended), deleting the `CMakeCache.txt` and `cmake` directory might help.
+
+## Build System
+
+The following questions are related to the build system.
+
+### What is CMake?
+
+`CMake` builds BuildSystems for different platforms, e.g. VisualStudio Solutions on Windows, Makefiles on Linux etc.
+This allows us to define in one central location (namely `CMakeLists.txt`) how OpenMS is build and have the platform specific stuff handled by `CMake`.
+View the [cmake website[(http://www.cmake.org) for more information.
+
+### How do I use CMake?
+
+See Installation instructions for your platform.
+In general, you call `CMake(.exe)` with some parameters to create the native build-system.
+Afterwards you can (but usually don't have to edit the current configuration using a GUI named `ccmake` (or `CMake-GUI` in Windows), which ships with `CMake`).
+
+> **_NOTE:_**  whenever `ccmake` is mentioned in this document, substitute this by `CMake-GUI` if your OS is Windows. You can also edit the `CMakeCache.txt` file directly.
+
+### How do I generate a build-system for Eclipse, KDevelop, CodeBlocks etc?
+
+Type `cmake` into a console. This will list the available code generators available on your platform, which you can pass to `CMake` using the `-G` option.
```

Comment:
> which you can pass to `CMake` using the `-G` option.

use them with `CMake` using the `-G` option.

Reply:
done